src/main/java/org/commcare/cases/instance/IndexedFixtureInstanceTreeElement.java

```diff
@@ -74,4 +78,37 @@ protected IndexedFixtureChildElement getChildTemplate() {
     public String getStorageCacheName() {
         return cacheKey;
     }
+
+    @Override
+    public int getAttributeCount() {
+        return attrHolder.getAttributeCount();
+    }
+
+    @Override
+    public String getAttributeNamespace(int index) {
+        return attrHolder.getAttributeNamespace(index);
+    }
+
+    @Override
+    public String getAttributeName(int index) {
+        return attrHolder.getAttributeName(index);
+    }
+
+    @Override
+    public String getAttributeValue(int index) {
+        return attrHolder.getAttributeValue(index);
+    }
+
+    @Override
+    public AbstractTreeElement getAttribute(String namespace, String name) {
+        TreeElement attr = attrHolder.getAttribute(namespace, name);
+        attr.setParent(this);
```

Comment:
Can you confirm that this method has no side effects that affect the current object (`this`)?

Reply:
The only affect is that [invalidates](https://github.com/dimagi/commcare-core/blob/master/src/main/java/org/javarosa/core/model/instance/TreeElement.java#L809) the reference cache for `this`. Is this something to be worried about ?

Reply:
No, invalidating the reference cache is fine, I just wanted to make sure that this wouldn't result in a "Backlink" being created where "this (the treelement part of the object)" now has an independent reference to "attr"

Reply:
got it. Doesn't look like that's happening.